modules/data.land/R/extract_soil_nc.R

```diff
@@ -0,0 +1,195 @@
+  ncvar <- list()
+  for(n in seq_along(soil.data)){
+    varname <- names(soil.data)[n]
+    if(length(soil.data[[n]])>1){
+      ## if vector, save by depth
+      ncvar[[n]] <- ncdf4::ncvar_def(name = varname, 
+                              units = soil.units(varname), 
+                              dim = depth)
+    }else {
+      ## else save as scalar
+      ncvar[[n]] <- ncdf4::ncvar_def(name = varname, 
+                              units = soil.units(varname), 
+                              dim=list())
+    }
+  }
+  
+  ## create new file
+  nc <- ncdf4::nc_create(new.file, vars = ncvar)
+  
+  ## add data
+  for (i in seq_along(ncvar)) {
+      ncdf4::ncvar_put(nc, ncvar[[i]], soil.data[[i]]) 
+  }
+
+  ncdf4::nc_close(nc)
+  
+  return(new.file)
+  
+}
+
+
+#' Get standard units for a soil variable
+#'
+#' @param varname 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' soil.units("soil_albedo")
+soil.units <- function(varname){
+  variables <- as.data.frame(matrix(c("soil_depth","m",
+                                      "soil_cec","meq/100g",
+                                      "volume_fraction_of_clay_in_soil","m3 m-3",
+                                      "volume_fraction_of_sand_in_soil","m3 m-3",
+                                      "volume_fraction_of_silt_in_soil","m3 m-3",
+                                      "volume_fraction_of_gravel_in_soil","m3 m-3",
+                                      "volume_fraction_of_water_in_soil_at_saturation","m3 m-3",
+                                      "volume_fraction_of_water_in_soil_at_field_capacity","m3 m-3",
+                                      "volume_fraction_of_condensed_water_in_dry_soil","m3 m-3",
+                                      "volume_fraction_of_condensed_water_in_soil_at_wilting_point","m3 m-3",
+                                      "soilC","percent",
+                                      "soil_ph","1",
+                                      "soil_bulk_density","kg m-3",
+                                      "soil_type","string",
+                                      "soil_hydraulic_b","1",
+                                      "soil_water_potential_at_saturation","m",
+                                      "soil_hydraulic_conductivity_at_saturation","m s-1",
+                                      "thcond0","W m-1 K-1",
+                                      "thcond1","W m-1 K-1",
+                                      "thcond2","1",
+                                      "thcond3","1",
+                                      "soil_thermal_conductivity","W m-1 K-1", 
+                                      "soil_thermal_conductivity_at_saturation","W m-1 K-1", 
+                                      "soil_thermal_capacity","J kg-1 K-1",
+                                      "soil_albedo","1"
+                                      ),
+                                    ncol=2,byrow = TRUE))
+  colnames(variables) <- c('var','unit')
```

Comment:
Since this is a tiny function that could get called a lot within the soil workflow, I think I was just trying to avoid the cost of constantly reloading the lookup table from disk every time the function is called.